compiler/fir/raw-fir/light-tree2fir/src/org/jetbrains/kotlin/fir/lightTree/converter/ExpressionsConverter.kt

```diff
@@ -823,9 +823,9 @@ class ExpressionsConverter(
         }
         val getArgument = context.arraySetArgument.remove(arrayAccess)
         return buildFunctionCall {
-            source = arrayAccess.toFirSourceElement()
+            source = (if (getArgument == null) arrayAccess else arrayAccess.getParent()!!).toFirSourceElement()
```

Comment:
Nit: You can introduce either `val isGet = getArgument == null` or `isSet` and use here and below for `name`.